pkg/quota/admission/clusterresourceoverride/admission.go

```diff
@@ -0,0 +1,176 @@
+package clusterresourceoverride
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"reflect"
+
+	oadmission "github.com/openshift/origin/pkg/cmd/server/admission"
+	configlatest "github.com/openshift/origin/pkg/cmd/server/api/latest"
+	"github.com/openshift/origin/pkg/project/cache"
+	"github.com/openshift/origin/pkg/quota/admission/clusterresourceoverride/api"
+	"github.com/openshift/origin/pkg/quota/admission/clusterresourceoverride/api/validation"
+	"k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/plugin/pkg/admission/limitranger"
+
+	"github.com/golang/glog"
+	"speter.net/go/exp/math/dec/inf"
+)
+
+const (
+	clusterResourceOverrideAnnotation = "quota.openshift.io/cluster-resource-override-enabled"
+	cpuBaseScaleFactor                = 1000.0 / (1024.0 * 1024.0 * 1024.0) // 1000 milliCores per 1GiB
+)
+
+func init() {
+	admission.RegisterPlugin(api.PluginName, func(client kclient.Interface, config io.Reader) (admission.Interface, error) {
+		return newClusterResourceOverride(client, config)
+	})
+}
+
+type internalConfig struct {
+	limitCPUToMemoryRatio     *inf.Dec
+	cpuRequestToLimitRatio    *inf.Dec
+	memoryRequestToLimitRatio *inf.Dec
+}
+type clusterResourceOverridePlugin struct {
+	*admission.Handler
+	internalConfig
```

Comment:
This looks like a structure that's parallel to `Config`.  Why do I have both?  Why not just `internalConfig`?